src/test/java/htsjdk/samtools/cram/encoding/external/ExternalCodecEquivalenceTest.java

```diff
@@ -0,0 +1,224 @@
+package htsjdk.samtools.cram.encoding.external;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.encoding.CRAMCodec;
+import htsjdk.samtools.cram.io.IOTestCases;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExternalCodecEquivalenceTest extends HtsjdkTest {
```

Comment:
This class needs a comment explaining the motivation for including it. Its essentially validating a property that several of the codecs happen to share, which is that for small values, the encoded streams are identical. But that property isn't exploited by, or even mentioned in the spec (?); it was only the *previous*, but incorrect encoding of the CF/MF data series incorrect that was accidentally exploiting it. Once this PR goes in, the incorrect implementation will be gone.

So I think its fine to retain the test, but it needs an explanation saying something like that its exploiting that property in order to cross validate the encoding implementations, or something like similar.